operators/src/source/gdal_source.rs

```diff
@@ -0,0 +1,456 @@
+
+        let grid_tile_provider = GdalSourceTileGridProvider {
+            global_pixel_size: global_size_in_pixels.into(),
+            tile_pixel_size: tile_size_in_pixels.into(),
+        };
+
+        let time_interval_provider = vec![TimeInterval::new_unchecked(1.into(),2.into())];
+        
+        let gdal_params = GdalSourceParameters{
+            base_path: "../operators/test-data/raster/srtm_38_03".into(),
+            file_name_with_time_placeholder: "srtm_38_03.tif".into(),
+            time_format: "".into(),
+            tick: None,
+            channel: None,
+        };
+
+        let gdal_source = GdalSource {
+            time_interval_provider,
+            grid_tile_provider,
+            gdal_params
+        };
+
+        let vres: Vec<Result<RasterTile2D<f32>, Error>> = gdal_source.time_tile_iter().map( |(time_interval, tile_information)| {gdal_source.load_tile_data(&time_interval, &tile_information)} ).collect();
+        assert_eq!(vres.len(), 6*6);
+        let upper_left_pixels: Vec<_> = vres.into_iter().map(
+            |t| {
+                let raster_tile = t.unwrap();
+                let tile_data = raster_tile.data;
+                tile_data.pixel_value_at_grid_index(&(0,0)).unwrap() // pixel
+                
+            }
+            ).collect();
+        assert_eq!(upper_left_pixels, srtm_tile_upper_left_pixel_values);
+    }
+
+    #[test]
+    fn test_tile_stream_len() {
+
+
+        let global_size_in_pixels = (6000, 6000);
+        let tile_size_in_pixels = (1000, 1000);
+        let srtm_tile_upper_left_pixel_values: Vec<f32> = vec![422.0, 434.0, 312.0, 347.0, 86.0, 231.0, 269.0, 184.0, 236.0, 360.0, 101.0, 199.0, 366.0, 334.0, 298.0, 162.0, 715.0, 799.0, 314.0, 253.0, 369.0, 333.0, 484.0, 619.0, 196.0, 651.0, 645.0, 1072.0, 3003.0, 1720.0, 179.0, 233.0, 1611.0, 2401.0, 1044.0, 1056.0];    
+
+        let grid_tile_provider = GdalSourceTileGridProvider {
+            global_pixel_size: global_size_in_pixels.into(),
+            tile_pixel_size: tile_size_in_pixels.into(),
+        };
+
+        let time_interval_provider = vec![TimeInterval::new_unchecked(1.into(),2.into())];
+        
+        let gdal_params = GdalSourceParameters{
+            base_path: "../operators/test-data/raster/srtm_38_03".into(),
+            file_name_with_time_placeholder: "srtm_38_03.tif".into(),
+            time_format: "".into(),
+            tick: None,
+            channel: None,
+        };
+
+        let gdal_source = GdalSource {
+            time_interval_provider,
+            grid_tile_provider,
+            gdal_params
+        };
+
+
+        let mut stream_data = block_on_stream(gdal_source.tile_stream::<f32>());
+
+        for p in srtm_tile_upper_left_pixel_values  {
+            let tile = stream_data.next().unwrap().unwrap();
+            let lue = tile.data.pixel_value_at_grid_index(&(0,0)).unwrap();
```

Comment:
As a suggestion, you could zip to iterators and compare the values.
But it is a good idea to compare the upper left values for now.

Reply:
I guess it is still better then hashes ;)